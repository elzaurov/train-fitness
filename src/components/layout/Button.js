import React from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity, StyleSheet, ActivityIndicator} from 'react-native';
import {Colors, Layout} from '../../constants';
import RegularText from './RegularText';

const Button = ({
  style,
  textStyle,
  children,
  primary,
  secondary,
  disabled,
  loading,
  ...props
}) => {
  const colorStyle = {backgroundColor: Colors.defaultButton};
  const disabledStyle = disabled ? {opacity: 0.5} : {};

  if (primary) {
    colorStyle.backgroundColor = Colors.primaryButton;
  } else if (secondary) {
    colorStyle.backgroundColor = Colors.secondaryButton;
  }

  let content;

  if (loading === true) {
    content = (
      <ActivityIndicator
        style={[styles.textStyle, textStyle]}
        color={Colors.white}
        size="small"
      />
    );
  } else {
    content =
      typeof children === 'string' ? (
        <RegularText style={[styles.textStyle, textStyle]}>
          {children.toUpperCase()}
        </RegularText>
      ) : (
        children
      );
  }

  return (
    <TouchableOpacity
      style={[styles.button, colorStyle, disabledStyle, style]}
      disabled={disabled}
      {...props}>
      {content}
    </TouchableOpacity>
  );
};

Button.propTypes = {
  style: PropTypes.any,
  textStyle: PropTypes.any,
  children: PropTypes.any,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
};

Button.defaultProps = {
  style: {},
  textStyle: {},
  children: undefined,
  primary: false,
  secondary: false,
  disabled: false,
  loading: false,
};

export default Button;

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 4,
  },
  textStyle: {
    padding: 8,
    paddingTop: Layout.padding,
    paddingBottom: Layout.padding,
    fontWeight: 'bold',
  },
});
