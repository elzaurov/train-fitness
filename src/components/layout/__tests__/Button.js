import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

test('Should render default Button', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render primary Button', () => {
  const tree = renderer.create(<Button primary />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render secondary Button', () => {
  const tree = renderer.create(<Button secondary />).toJSON();
  expect(tree).toMatchSnapshot();
});
