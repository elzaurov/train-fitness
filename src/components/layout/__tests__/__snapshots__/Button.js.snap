// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render default Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1ab15d",
      "borderRadius": 2,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
      "width": 718,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;

exports[`Should render primary Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#830d0d",
      "borderRadius": 2,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
      "width": 718,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;

exports[`Should render secondary Button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d9252b",
      "borderRadius": 2,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
      "width": 718,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;
